src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -35,6 +35,215 @@ public static partial class Base64
         ///   or if the input is incomplete (i.e. not a multiple of 4) and <paramref name="isFinalBlock"/> is <see langword="true"/>.
         /// </returns>
         public static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> bytes, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
+        {
+            // Validation must occur prior to decoding as the actual length will impact future calculations
+            bool containsIgnoredBytes = utf8.IndexOfAny(BytesToIgnore) != -1;
```

Comment:
I eject this is going to regress performance, which we want to avoid. It should be possible to structure this such that we only pay additional cost in the existing implementation when it encounters an invalid character according to its current definition, which includes whitespace, at which point it can fall back to an implementation that allows for whitespace but is slower.